src/acceleration/impl/ResidualSumPreconditioner.hpp

```diff
@@ -17,7 +17,7 @@ namespace impl {
  */
 class ResidualSumPreconditioner : public Preconditioner {
 public:
-  ResidualSumPreconditioner(int maxNonConstTimeWindows);
+  ResidualSumPreconditioner(int maxNonConstTimeWindows, bool = true, double updatePreconLimit = 10.0);
```

Comment:
```suggestion
  ResidualSumPreconditioner(int maxNonConstTimeWindows, bool preconForceUpdate = true, double updatePreconLimit = 10.0);
```
??
Why do you need the defaults here? Aren't these covered in the configuration already?